### ISSUE TITLE
Typescript to Python JSII CDK Construct Interface issue

### ISSUE THREAD

**Issue by zARODz11z** [aws/jsii#2749]
### Affected Languages

- [x] `TypeScript` or `Javascript`
- [x] `Python`
- [ ] `Java`
- [ ] .NET (`C#`, `F#`, ...)
- [ ] `Go`

### General Information
* **JSII Version: 1.24.0**
* **Platform: Mac OS** 

### The Question
I converted an existing AWS CDK Construct Library to use Projen & JSII, at the moment I want to target NPM and PyPi for releases of the Construct Library. When using JSII I got a compilation error for an interface I was using to hold properties during the initialization of my Construct so I set it to a behavioral interface by prefacing it with an 'I'. The Javascript version of the Construct takes in a scope, id, and props which are configurations the user can setup and they will be used later in the Construct code. I was able to get JSII to produce the Python artifacts and published it to TestPyPi. I began trying to test my Python CDK Construct produced by JSII but am running into issues with instantiating my Construct. When I attempt to pass in the props my cdk synth fails saying I got an unexpected keyword argument. The constructor went from this in Typescript:
```
  constructor(scope: cdk.Construct, id: string, props: DatadogProps) {
    super(scope, id);
    this.scope = scope;
    this.props = props;
    validateProps(this.props);
    this.transport = new Transport(
      this.props.flushMetricsToLogs,
      this.props.site,
      this.props.apiKey,
      this.props.apiKMSKey,
      this.props.extensionLayerVersion,
    );
  }
```
To this in Python:
```
def __init__(
        self,
        scope: aws_cdk.core.Construct,
        id: builtins.str,
        props: "IDatadogProps",
    ) -> None:
        '''
        :param scope: -
        :param id: -
        :param props: -
        '''
        jsii.create(Datadog, self, [scope, id, props])
```

To sum it up I created a Python artifact via JSII based on a Typescript CDK Construct and published it to TestPyPi, I pulled that package down to test it in a CDK stack but the constructor is not working like it does in the NPM package. Im not sure if I am just interacting with my Python CDK Construct incorrectly or if I need to change something in the Typescript project that the Python Package originated from. If anyone needs more details feel free to ask! Thank you in advance.

**Comment by zARODz11z** [comment]
Im bumping this to see if anyone can help, let me know if you need more details.

**Comment by zARODz11z** [comment]
This was the setter generated by JSII for the interface, i'm not sure why the "..." were included: 
```
    @node_layer_version.setter
    def node_layer_version(self, value: typing.Optional[jsii.Number]) -> None:
        ...
```

**Comment by zARODz11z** [comment]
I am trying to set the properties using this setter that JSII provided: 
Setter I am trying to invoke (in python)
```
@props.setter
    def props(self, value: "IDatadogProps") -> None:
        jsii.set(self, "props", value)
```
Code I am invoking the setter with
```
        props = {
            'node_layer_version': 49
        }
        datadog = Datadog(self, "Datadog", Datadog.props.setter(jsii.create("Object", props)))
```
Error I am receiving:
```
fqn=klass.__jsii_type__ or "Object",
AttributeError: 'str' object has no attribute '__jsii_type__'
Subprocess exited with error 1
```
Can anyone explain what I am doing wrong?

**Comment by zARODz11z** [comment]
Changing from a behavioral interface to a struct fixed my issue https://aws.github.io/jsii/user-guides/lib-author/typescript-restrictions/#interfaces
